src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -674,10 +695,14 @@ int TieredHNSWIndex<DataType, DistType>::addVector(const void *blob, labelType l
     int ret = 1;
     auto hnsw_index = this->getHNSWIndex();
     if (this->getWriteMode() == VecSim_WriteInPlace) {
+        // First, check if we need to overwrite the vector in-place for single (from both indexes).
+        if (!this->backendIndex->isMultiValue()) {
+            ret -= this->deleteVector(label);
+        }
+        // Insert the vector to the HNSW index. Internally, we will never have to overrite the label
+        // since we already checked it outside.
```

Comment:
As discussed - we are checking it here (line 700), and it is required to handle deletion in place in case of overwrites (instead of letting HNSW handle the overwrite without invalidating the jobs)